backend/substrapp/compute_tasks/image_builder.py

```diff
@@ -148,18 +162,16 @@ def _assert_dockerfile_exist(dockerfile_path):
         raise compute_task_errors.BuildError(f"Dockerfile does not exist : {dockerfile_fullpath}")
 
 
-def _log_container_logs(pod_name: str) -> None:
+def _get_container_logs(pod_name: str, ignore_errors: bool = False) -> str:
     kubernetes.config.load_incluster_config()
     k8s_client = kubernetes.client.CoreV1Api()
 
-    container_logs = get_pod_logs(k8s_client, name=pod_name, container=KANIKO_CONTAINER_NAME)
-    if isinstance(container_logs, bytes):
-        logs = [log for log in container_logs.decode().split("\n")]
-    else:
-        logs = [log for log in container_logs.split("\n")]
-
-    for log in logs:
-        logger.info(log, pod_name=pod_name)
+    try:
+        return get_pod_logs(k8s_client, name=pod_name, container=KANIKO_CONTAINER_NAME)
+    except Exception:
```

Comment:
I took another approach in https://github.com/Substra/substra-backend/pull/483/commits/22bbbead059076c69a618af94d09c8336547db31

I added the `ignore_pod_not_found` parameter because, as a utils function, `get_pod_logs` shouldn't ignore errors _by default_. Actually, as you mentioned in your other comment `get_pod_logs` ignores errors by default currently in the main branch, which mislead me into adding an unneeded try/catch. Said another way, It can be dangerous to swallow errors by default without explicit caller request.

Let me know what you think.

Reply:
It looks good.